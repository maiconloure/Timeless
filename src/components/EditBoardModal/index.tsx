/* eslint-disable @typescript-eslint/no-unused-vars */
import { Input, Button } from 'capstone-project';
import React from 'react';

import * as Interface from '../../redux/actions/interface.action';
import { icons } from '../../utils/importAll';
import { EditBoardModalProps } from '../ComponentsInterface';
import * as St from '../EditModalStyles';
import ModalBackground from '../ModalBackground';

const EditBoardModal = ({
  boards,
  values: { boardTitle, boardDescription },
  showModal: { showMobileMenu, showEditModal, showBoardModal, setShowEditModal, setShowBoardModal },
  handlers: {
    handlerTitle,
    handlerDescription,
    handleReturnForm,
    handlerModifyBoard,
    handlerRemoveBoard,
    handlerSelectBoard,
    handlerCreateBoard,
    handlerSubmitForm,
  },
}: EditBoardModalProps) => (
  <>
    <ModalBackground show={showBoardModal} />
    <St.CardModal
      icon={icons.closeWindow}
      title="Boards"
      data={[showBoardModal, setShowBoardModal]}
      styles={{
        size: showMobileMenu ? 'normal' : 'large',
        fontSize: showMobileMenu ? 'large' : 'largest',
        bgColorPrimary: '#cdd9e2',
        colorPrimary: '#014d82',
      }}>
      <>
        {showEditModal ? (
          <St.Form>
            <Button onClick={handleReturnForm}>Voltar</Button>
            {!showMobileMenu && <h2>Titulo</h2>}
            <Input
              type="text"
              placeholder={boardTitle}
              width="220px"
              fontSize="2rem"
              height="40px"
              onTextChange={handlerTitle}
            />
            {!showMobileMenu && <h2>Descrição</h2>}
            <Input
              type="text"
              placeholder={boardDescription}
              width="220px"
              fontSize="2rem"
              height="40px"
              onTextChange={handlerDescription}
            />
            <Button fontSize="2.6rem" height="44px" weight={600} onClick={handlerSubmitForm}>
              Modificar
            </Button>
          </St.Form>
        ) : (
          <>
            <St.ModalContent>
              <St.MenuModal>
                <St.CardModalDescription>Novo Board</St.CardModalDescription>

                <St.CardModalButton onClick={handlerCreateBoard}>Criar</St.CardModalButton>
              </St.MenuModal>
            </St.ModalContent>

            {boards &&
              boards.map((board: Interface.UserBoards, key: number) => (
                <St.ModalContent key={key}>
                  <h2>{board.title}</h2>

                  <St.CardModalSection>
                    <St.CardModalDescription>{board.description}</St.CardModalDescription>

                    <div>
                      <St.CardModalButton onClick={() => handlerSelectBoard(board)}>
                        Selecionar
                      </St.CardModalButton>

                      <St.CardModalButton onClick={() => handlerModifyBoard(board)}>
                        Modificar
                      </St.CardModalButton>

                      <St.CardModalButton onClick={() => handlerRemoveBoard(board)}>
                        Remover
                      </St.CardModalButton>
                    </div>
                  </St.CardModalSection>
                </St.ModalContent>
              ))}
          </>
        )}
      </>
    </St.CardModal>
  </>
);

export default EditBoardModal;
