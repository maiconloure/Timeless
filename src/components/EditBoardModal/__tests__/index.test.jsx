import EditBoardModal from '../index';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Snapshot', () => {
  it('renders', () => {
    const tree = renderer
      .create(
        <EditBoardModal
          boards={['test']}
          values={{ boardTitle: 'title', boardDescription: 'description' }}
          showModal={{ showEditModal: true, showBoardModal: true, setShowBoardModal: jest.fn() }}
          handlers={{
            handlerTitle: jest.fn(),
            handlerDescription: jest.fn(),
            handleReturnForm: jest.fn(),
            handlerModifyBoard: jest.fn(),
            handlerRemoveBoard: jest.fn(),
            handlerSelectBoard: jest.fn(),
            handlerCreateBoard: jest.fn(),
            handlerSubmitForm: jest.fn(),
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders but showBoardModal is false', () => {
    const tree = renderer
      .create(
        <EditBoardModal
          boards={['test']}
          values={{ boardTitle: 'title', boardDescription: 'description' }}
          showModal={{ showEditModal: true, showBoardModal: false, setShowBoardModal: jest.fn() }}
          handlers={{
            handlerTitle: jest.fn(),
            handlerDescription: jest.fn(),
            handleReturnForm: jest.fn(),
            handlerModifyBoard: jest.fn(),
            handlerRemoveBoard: jest.fn(),
            handlerSelectBoard: jest.fn(),
            handlerCreateBoard: jest.fn(),
            handlerSubmitForm: jest.fn(),
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('not renders', () => {
    const tree = renderer
      .create(
        <EditBoardModal
          boards={['test']}
          values={{ boardTitle: 'title', boardDescription: 'description' }}
          showModal={{ showEditModal: false, showBoardModal: false, setShowBoardModal: jest.fn() }}
          handlers={{
            handlerTitle: jest.fn(),
            handlerDescription: jest.fn(),
            handleReturnForm: jest.fn(),
            handlerModifyBoard: jest.fn(),
            handlerRemoveBoard: jest.fn(),
            handlerSelectBoard: jest.fn(),
            handlerCreateBoard: jest.fn(),
            handlerSubmitForm: jest.fn(),
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
