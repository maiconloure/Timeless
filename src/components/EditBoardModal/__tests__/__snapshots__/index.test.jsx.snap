// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot not renders 1`] = `
<div
  className="sc-fKgIGh cNCOQf"
>
  <div
    className="sc-crzoUp inpdim"
    onClick={[Function]}
  >
    <img
      alt="close icon"
    />
  </div>
  <h1
    className="sc-bCwgka dJmVvq"
  >
    Boards
  </h1>
  <div
    className="sc-iwaifL sc-cxNIbT fFSxYx hcDzte"
  >
    <div
      className="sc-ArjOu dRKcKi"
    >
      <div
        className="sc-jNnnWF lbALrE"
      >
        <p
          className="sc-khIgXV hVcsEV"
        >
          Novo Board
        </p>
        <button
          className="sc-dPaNSN cmJrtY"
          onClick={[MockFunction]}
        >
          Criar
        </button>
      </div>
    </div>
    <div
      className="sc-ArjOu dRKcKi"
    >
      <h2 />
      <div
        className="sc-bBjRzc kCwEgw"
      >
        <p
          className="sc-khIgXV hVcsEV"
        />
        <div>
          <button
            className="sc-dPaNSN cmJrtY"
            onClick={[Function]}
          >
            Selecionar
          </button>
          <button
            className="sc-dPaNSN cmJrtY"
            onClick={[Function]}
          >
            Modificar
          </button>
          <button
            className="sc-dPaNSN cmJrtY"
            onClick={[Function]}
          >
            Remover
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot renders 1`] = `
Array [
  <div
    className="sc-efHXLn bNzgxe"
  />,
  <div
    className="sc-fKgIGh iYqHxB"
  >
    <div
      className="sc-crzoUp inpdim"
      onClick={[Function]}
    >
      <img
        alt="close icon"
      />
    </div>
    <h1
      className="sc-bCwgka dJmVvq"
    >
      Boards
    </h1>
    <div
      className="sc-iwaifL sc-cxNIbT fFSxYx hcDzte"
    >
      <div
        className="sc-cOigif haWtsF"
      >
        <button
          className="sc-bdnylx hhyBUP"
          color="#014D82"
          fontSize="1.4rem"
          height="55px"
          onClick={[MockFunction]}
          radius="6px"
          width="170px"
        >
          Voltar
        </button>
        <div
          className="sc-dlnjPT"
        >
          <input
            className="sc-hKFyIo evVGth"
            color="#014D82"
            fontSize="2rem"
            height="40px"
            name="fname"
            onChange={[Function]}
            placeholder="title"
            type="text"
            width="220px"
          />
        </div>
        <div
          className="sc-dlnjPT"
        >
          <input
            className="sc-hKFyIo evVGth"
            color="#014D82"
            fontSize="2rem"
            height="40px"
            name="fname"
            onChange={[Function]}
            placeholder="description"
            type="text"
            width="220px"
          />
        </div>
        <button
          className="sc-bdnylx jmFhMj"
          color="#014D82"
          fontSize="2.6rem"
          height="44px"
          onClick={[MockFunction]}
          radius="6px"
          width="170px"
        >
          Modificar
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Snapshot renders but showBoardModal is false 1`] = `
<div
  className="sc-fKgIGh cNCOQf"
>
  <div
    className="sc-crzoUp inpdim"
    onClick={[Function]}
  >
    <img
      alt="close icon"
    />
  </div>
  <h1
    className="sc-bCwgka dJmVvq"
  >
    Boards
  </h1>
  <div
    className="sc-iwaifL sc-cxNIbT fFSxYx hcDzte"
  >
    <div
      className="sc-cOigif haWtsF"
    >
      <button
        className="sc-bdnylx hhyBUP"
        color="#014D82"
        fontSize="1.4rem"
        height="55px"
        onClick={[MockFunction]}
        radius="6px"
        width="170px"
      >
        Voltar
      </button>
      <div
        className="sc-dlnjPT"
      >
        <input
          className="sc-hKFyIo evVGth"
          color="#014D82"
          fontSize="2rem"
          height="40px"
          name="fname"
          onChange={[Function]}
          placeholder="title"
          type="text"
          width="220px"
        />
      </div>
      <div
        className="sc-dlnjPT"
      >
        <input
          className="sc-hKFyIo evVGth"
          color="#014D82"
          fontSize="2rem"
          height="40px"
          name="fname"
          onChange={[Function]}
          placeholder="description"
          type="text"
          width="220px"
        />
      </div>
      <button
        className="sc-bdnylx jmFhMj"
        color="#014D82"
        fontSize="2.6rem"
        height="44px"
        onClick={[MockFunction]}
        radius="6px"
        width="170px"
      >
        Modificar
      </button>
    </div>
  </div>
</div>
`;
